devmand/gateway/src/devmand/channels/cli/CliFlavour.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2019-present, Facebook, Inc.
+// All rights reserved.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree. An additional grant
+// of patent rights can be found in the PATENTS file in the same directory.
+
+#include <boost/algorithm/string/trim.hpp>
+#include <devmand/channels/cli/CliFlavour.h>
+#include <regex>
+
+namespace devmand {
+namespace channels {
+namespace cli {
+
+using devmand::channels::cli::CliFlavour;
+using devmand::channels::cli::CliInitializer;
+using devmand::channels::cli::PromptResolver;
+using devmand::channels::cli::sshsession::SshSessionAsync;
+using devmand::channels::cli::EmptyInitializer;
+using devmand::channels::cli::UbiquitiInitializer;
+using devmand::channels::cli::DefaultPromptResolver;
+
+static const int DEFAULT_MILLIS = 1000;
+
+void EmptyInitializer::initialize(shared_ptr<SshSessionAsync> session) {
+    (void) session;
+}
+
+void UbiquitiInitializer::initialize(shared_ptr<SshSessionAsync> session) {
+  session->write("enable\n")
```

Comment:
Yes, we want to make this async. But we want to do that after we have our Keepalive/Reconnect code ready, because it might tie into that. 

So making the initialization of a Cli device async is in our backlog ... and we will look at that before December